internal/rbd/controllerserver.go

```diff
@@ -589,6 +591,12 @@ func (cs *ControllerServer) createVolumeFromSnapshot(
 
 		return err
 	}
+	err = rbdVol.unsetAllMetadata(k8s.GetSnapshotMetadataKeys())
```

Comment:
not related to this PR I see a cleanup bug here. if the unsetAllMetadata or copyEncryptionConfig or expand fails we need to remove the clone. can you please open an issue for this one?

Reply:
Sure Madhu

Reply:
https://github.com/ceph/ceph-csi/issues/3103